alphagov/paas-release-ci
# [#130188061] Add pipeline for building RDS broker bosh releases

## What

We want a pipeline which builds releases of our RDS broker every time a pull request is raised and every time code is merged into master.

This pull request introduces two pipelines:

* one which creates a bucket for releases to be stored in, and dynamically generates other pipelines
* a pipeline for building RDS broker bosh releases

Currently the 'dynamically' generated pipelines aren't so dynamic because there is only one of them and the name is hard coded, but this can be refactored when we add more bosh releases.

## How to review

Review in conjunction with https://github.com/alphagov/paas-pullrequest-resource/pull/1

 * Use the paas-bootstrap repository to create a bootstrap:

    ```
    git fetch && git checkout master
    # Set DEPLOY_ENV to a value different to any existing deployments!
    BOSH_INSTANCE_PROFILE=bosh-director-build CONCOURSE_INSTANCE_PROFILE=concourse-build DEPLOY_ENV=$DEPLOY_ENV make dev bootstrap
    ```
* Run the bootstrap `create` pipeline.
* Navigate to this repository and checkout this branch (`rds-broker-release-pipeline`)
* Run `BRANCH=rds-broker-release-pipeline make dev pipelines` then manually trigger the setup pipeline. Concourse should then have an rds-broker pipeline. (You may need to reload the page before it appears in the side menu)
* Test the rds-broker pipeline. I will leave the precise details to the reviewer, but the aim of the pipeline is to:
    * trigger the `build-dev-release` job every time the RDS broker bosh release repository has a new or updated pull request
    * **NOT** trigger the `build-dev-release` for pull requests created from forked repos.
    * trigger the `build-final-release` jobs every time new commits are added to the master branch

## Before merging

* https://github.com/alphagov/paas-pullrequest-resource/pull/1 must be merged (and allowed to build on dockerhub)
* The `[TMP]` commit on this branch pointing the resource at that branch must be removed.

## After merge

We will deploy this to CI once it's been merged.

## Who

Anyone but me or @alext
